boot/bootutil/src/loader.c

```diff
@@ -1103,6 +1082,8 @@ boot_validate_slot(struct boot_loader_state *state, int slot,
 
 #if defined(MCUBOOT_OVERWRITE_ONLY) && defined(MCUBOOT_DOWNGRADE_PREVENTION)
     if (slot != BOOT_PRIMARY_SLOT) {
+        int rc;
```

Comment:
define above so it is visible for whole function to avoid shadow

Reply:
I have removed the one from the main scope and this is the only instance left.

Reply:
this now breaks https://github.com/mcu-tools/mcuboot/blob/d712686e2846e17bad9798d5417377d47f428c76/boot/bootutil/src/loader.c#L1138C1-L1138C82